### PR TITLE
feat: add covariance matrix of tracks to ROOT output

### PR DESCRIPTION
**Pull request by owtscharenko**
This PR integrates the feature request from #2453 . A boolean switch `writeCovMat` is added to the `RootTrajectorySummaryWriter`. If the switch is `true`, the 6x6 covariance matrix of fitted tracks is written to the ROOT output file as one branch for every entry of the matrix.

### CODE REVIEW

**Review by benjaminhuth — COMMENTED** [review]
In principle looks good to me!
Only a few comments:
* The variances in principle are now written twice to the file (they are also in the `eLOC0_err` etc.), but this is ok I think since this is entirely optional
* The covariance is first copied into a `TMatrixD`, even if the switch is off... I think it would be more efficient to copy them directly from the `boundParam.covariance()`. 
* If there is no covariance associated to the parameters, the unit matrix is written to the file... Maybe it would be better to write some invalid values like -1?

I could imagine a construct like:

```c++ 
if (m_cfg.writeCovMat == true) { 
	auto getCov = [&](auto i, auto j) { return boundParam.covariance() ? boundParam.covariance().value()(i, j) : -1; };

    m_cov_eLOC0_eLOC0.push_back(getCov(0,0));
    // ...
}
```

**Comment by owtscharenko** [comment]
I agree, it makes sense to write invalid default values to the matrix.
I also agree that your construct is much more elegant I can change the code accordingly.
Thank you for the input

**Review by felix-russo — COMMENTED** [review]
I just saw that @benjaminhuth already reviewed sorry!

Like Benjamin, I don't think we should save a unit matrix if no covariance matrix is available. I am not a root expert, but is there a possibility to either 

- not fill the branch or 
- provide a None value to it 

in that case? "Just" adding some -1 might look strange in the histograms I guess

File: Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp [hunk 1]
```diff
@@ -150,6 +195,7 @@ ActsExamples::RootTrajectorySummaryWriter::finalize() {
   m_outputTree->Write();
   m_outputFile->Close();
 
+  ACTS_INFO("Wrote full covariance matrix to tree");
```
Comment (benjaminhuth) [inline Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp:198]: This probably should only be printed if the option is actually `true`, right?

**Comment by benjaminhuth** [comment]
> in that case? "Just" adding some -1 might look strange in the histograms I guess

Yeah, I'm also not sure what is the best way here. But I just saw that we actually already have a standard a way to express this by using `NaNfloat` (this is used in this writer as defaults for other potentially optional values. 

We should stick to that I think.

**Comment by paulgessinger** [comment]
@felix-russo Since it's floats we could write `NaN`, in principle.

File: Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp (hunk 1 above)
Comment (owtscharenko) [reply Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp:198]: yes, sorry for that.

Comment (owtscharenko) [reply Examples/Io/Root/src/RootTrajectorySummaryWriter.cpp]: I changed that in https://github.com/owtscharenko/acts/commit/46e8bce2a9b9c9691b3ceec40be4053b04cbb264

**Comment by owtscharenko** [comment]
I implemented the suggested lambda statement, however I needed to draw the covariance from `traj.trackParameters` instead of `boundParam` since latter is only defined in the if statement in L. 398

**Review by benjaminhuth — APPROVED** [review]
Looks good to me! Let's get it in!

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625625) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625624) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625623) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625622) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625621) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625620) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625619) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625618) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625617) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625616) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625615) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625614) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625613) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625612) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625611) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625610) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32625609) |

**Pull request merged**